src/SoftWire.h

```diff
@@ -7,6 +7,32 @@
 #include <stdint.h>
 #include <AsyncDelay.h>
 
+//
+// 2022-04-13 by Technik Gegg
+//
+//    On the Arduino Core STM32 framework pin definitions have moved
+//    from uint8_t to uint32_t in order to reflect higher pin numbers on 
+//    some of the newer MCUs available. 
+//    Using uint8_t for storing pin numbers internally may cause cut offs and thus,
+//    such pins will not be addressed and function correctly.
+//    For other frameworks, which do not support newer MCUs, this isn't an issue
+//    and hence I'm using the conditional typedef down below.
+//
+//    If your project or library bases on/supports the Arduino Core STM32 framework,
+//    I'd highly recommend using the same construct as shown below in every location
+//    of your source code where you either store or return pin numbers, just to be future proof.
+//
+//    Using 32 bit for storing pin numbers might seem a bit overkill but one
+//    never knows what the future brings and eventually, using a specific type
+//    allows to adopt your source code quickly for a different framework without the 
+//    hassle of going through the code and changing/casting those pin definitions.
+//
+#if !defined(STM32_CORE_VERSION)
```

Comment:
This can be simplified to remove the logical NOT and swap the if/else blocks.

Reply:
So much simpler now ;o)